src/generator/AutoRest.Ruby/Templates/ServiceClientTemplate.cshtml

```diff
@@ -79,6 +79,62 @@ module @Settings.Namespace
     end
 
     @EmptyLine
+    #
+    # Makes a request and returns the body of the response
+    # @@param method [Symbol] with any of the following values :get, :put, :post, :patch, :delete   
+    # @@param path [String] the path, relative to {api_endpoint}
+    # @@param options [Hash{String=>String}] specifying any request options like :body
+    # @@return [Hash{String=>String}] containing the body of the response
+    #
+    def make_request(method, path, options = {})
+      result = make_request_async(method, path, options).value!
+      result.response.body.to_s.empty? ? nil : JSON.load(result.response.body)
+    end
+    @EmptyLine
+
+    #
+    # Makes a request asynchronously
+    # @@param method [Symbol] with any of the following values :get, :put, :post, :patch, :delete
+    # @@param path [String] the path, relative to {api_endpoint}
+    # @@param options [Hash{String=>String}] specifying any request options like :body
+    # @@return [Concurrent::Promise] Promise object which holds the HTTP response.
+    #
+    def make_request_async(method, path, options = {})
```

Comment:
In the ServiceClientTemplate method signature, I'm removing request_url/base_url parameter from the signature, because we know that one there, so the user doesn't need to pass it. For this reason, I'm leaving it as is.